src/bin/pg_autoctl/keeper.c

```diff
@@ -171,6 +171,17 @@ keeper_should_ensure_current_state_before_transition(Keeper *keeper)
 		return false;
 	}
 
+	if (keeperState->current_role == SECONDARY_STATE &&
```

Comment:
After this change, we won't be getting into `keeper_ensure_current_state()` and hence not calling `keeper_maintain_replication_slots()`, that might be OK because we call it on `catching up-> secondary` transition anyway. 

If so, do we still need the `case SECONDARY_STATE:` in `keeper_ensure_current_state`?

Also, on `keeper_fsm_step` we call `keeper_ensure_current_state()` without `keeper_should_ensure_current_state_before_transition()` check. Is that intended?

Reply:
We still ensure the current state when the current state and assign state are the same, no change there. The change is about ensure current state before implementing an FSM transition only: when the state is asked to change.

Reply:
We call `keeper_ensure_current_state` in two places in the main keeper loop. One place is when there is a state transition to implement, and then sometimes we want to ensure the current state before implementing the transition (that's mostly when everything is under control). The other place is when the current and assigned state are the same, which is most of the times really.

This PR doesn't change `keeper_ensure_current_state`, it only makes it so that when we are asked to implement a transition to another state, we sometimes don't ensure the current state first. On the primary that's already what we did with the exceptions about DRAINING, DEMOTED, and DEMOTE_TIMEOUT states. It turns out we need the same kind of provisions for some cases on the secondary node(s).